File: backend/accounts/services/permission.py
from core.base import BaseCrudService

from accounts.models import Permission


class PermissionService(BaseCrudService):
    """
    Service métier des permissions.
    """

    model = Permission

    @classmethod
    def create(cls, data):

        return super().create(
            module=data["module"],
            resource=data["resource"],
            action=data["action"],
            name=data["name"],
            description=data.get("description", ""),
            is_active=data.get("is_active", True),
        )

    @classmethod
    def update(cls, permission, data):

        return super().update(
            permission,
            module=data.get("module", permission.module),
            resource=data.get("resource", permission.resource),
            action=data.get("action", permission.action),
            name=data.get("name", permission.name),
            description=data.get(
                "description",
                permission.description,
            ),
            is_active=data.get(
                "is_active",
                permission.is_active,
            ),
        )

File: frontend/src/services/permission.js
import { api } from "./api";

export const permissionService = {

    async getAll() {

        return await api.get("/auth/permissions/");

    },

    async create(data) {

        return await api.post("/auth/permissions/", data);

    },

    async update(id, data) {

        return await api.put(`/auth/permissions/${id}/`, data);

    },

    async delete(id) {

        return await api.delete(`/auth/permissions/${id}/`);

    },

};